src/modules/commander/esc_calibration.cpp

```diff
@@ -105,14 +92,14 @@ int do_esc_calibration(orb_advert_t *mavlink_log_pub, struct actuator_armed_s* a
 #else
 	int	fd = -1;
 
-	struct	battery_status_s battery;
+	struct	battery_status_s battery = {};
 	int	batt_sub = -1;
 	bool	batt_updated = false;
-	bool	batt_connected = false;
+	bool	batt_connected = true;	// for safety resons assume battery is connected, will be cleared below if not the case
 
-	hrt_abstime battery_connect_wait_timeout = 30000000;
-	hrt_abstime pwm_high_timeout = 3000000;
-	hrt_abstime timeout_start;
+	hrt_abstime battery_connect_wait_timeout = 30*1000*1000;
+	hrt_abstime pwm_high_timeout = 3*1000*1000;
+	hrt_abstime timeout_start = 0;
```

Comment:
Why do we need special behaviour for ocpoc here? Is actuator_armed.in_esc_calibration_mode not being correctly by commander? These type of scattered platform differences are horrible.

Reply:
@dagar I was wondering as well when I looked at that code. Seems like it keeps the high pwm pulse for 60 seconds- seems like a very long time.